kotest-assertions-arrow-fx-coroutines/src/commonMain/kotlin/io/kotest/assertions/arrow/fx/coroutines/Resource.kt

```diff
@@ -1,178 +1,9 @@
 package io.kotest.assertions.arrow.fx.coroutines
 
-import arrow.core.NonEmptyList
-import arrow.core.None
-import arrow.core.Option
-import arrow.core.Some
-import arrow.core.ValidatedNel
-import arrow.core.continuations.AtomicRef
-import arrow.core.continuations.update
-import arrow.core.identity
-import arrow.core.invalidNel
-import arrow.core.prependTo
-import arrow.core.traverse
-import arrow.core.valid
-import arrow.fx.coroutines.ExitCase
-import io.kotest.core.TestConfiguration
-import arrow.fx.coroutines.Platform
 import arrow.fx.coroutines.Resource
-import arrow.fx.coroutines.bracketCase
-import arrow.fx.coroutines.continuations.ResourceScope
-import io.kotest.core.listeners.TestListener
-import io.kotest.core.spec.Spec
-import kotlin.properties.ReadOnlyProperty
-import kotlin.reflect.KProperty
-import kotlinx.coroutines.Dispatchers
 
 public suspend fun <A> Resource<A>.shouldBeResource(a: A): A =
   use { it shouldBe a }
 
 public suspend fun <A> Resource<A>.shouldBeResource(expected: Resource<A>): A =
   zip(expected).use { (a, b) -> a shouldBe b }
-
-/**
- * ensures that all finalizers of [resource] are invoked for the given [TestConfiguration],
- * by registering a listener.
- *
- * ```kotlin
- * interface Database { // given a Dependency
- *   fun isRunning(): Boolean
- *   suspend fun version(): String?
- * }
- *
- * // and a Way to create it, here with Hikari and SqlDelight
- * fun database(hikari: HikariDataSource): Database =
- *   object : Database {
- *     override fun isRunning(): Boolean = hikari.isRunning
- *     override suspend fun version(): String? =
- *       hikari.queryOneOrNull("SHOW server_version;") { string() }
- *    }
- *
- * fun hikari(): Resource<HikariDataSource> =
- *   Resource.fromCloseable {
- *     HikariDataSource(
- *      HikariConfig().apply {
- *       // config settings
- *      }
- *     )
- *   }
- * fun sqlDelight(hikariDataSource: HikariDataSource): Resource<SqlDelight> =
- *   Resource.fromCloseable { hikariDataSource.asJdbcDriver() }.map { driver -> SqlDelight(driver) }
- *
- * // we can define a Database Resource
- * fun database(): Resource<Database> =
- *  hikari().flatMap { hikari ->
- *   sqlDelight(hikari).map { sqlDelight -> database(hikari) }
- *  }
- *
- * // and define Tests with our backend Application, here with Ktor
- * class HealthCheckSpec :
- *   StringSpec({
- *     val database: Database by resource(database())
- *
- *     "healthy" {
- *       testApplication {
- *         application { app(database) }
- *         val response = client.get("/health")
- *         response.status shouldBe HttpStatusCode.OK
- *         response.bodyAsText() shouldBe Json.encodeToString(HealthCheck("14.1"))
- *       }
- *     }
- *   })
- * ```
- */
-public fun <A> TestConfiguration.resource(resource: Resource<A>): ReadOnlyProperty<Any?, A>  =
```

Comment:
Seems like a breaking change to drop this? Would probably be more convenient for users to `@Deprecate` and remove later?

Reply:
Aha, nice! 😄